File: ru.kmorozov.library.data.server/src/main/java/ru/kmorozov/library/data/server/condition/StorageEnabledCondition.java
package ru.kmorozov.library.data.server.condition;

import org.springframework.context.annotation.Condition;
import org.springframework.context.annotation.ConditionContext;
import org.springframework.core.type.AnnotatedTypeMetadata;

public class StorageEnabledCondition implements Condition {

    @Override
    public boolean matches(final ConditionContext conditionContext, final AnnotatedTypeMetadata annotatedTypeMetadata) {
        return Boolean.valueOf(conditionContext.getEnvironment().getProperty("rest.controllers.storage.enabled"));
    }
}

File: ru.kmorozov.library.data.server/src/main/java/ru/kmorozov/library/data/server/condition/StorageEnabledCondition.kt
package ru.kmorozov.library.data.server.condition

import org.springframework.context.annotation.Condition
import org.springframework.context.annotation.ConditionContext
import org.springframework.core.type.AnnotatedTypeMetadata

class StorageEnabledCondition : Condition {

    override fun matches(conditionContext: ConditionContext, annotatedTypeMetadata: AnnotatedTypeMetadata): Boolean {
        return java.lang.Boolean.valueOf(conditionContext.environment.getProperty("rest.controllers.storage.enabled"))
    }
}
